src/disk_io_thread_pool.cpp

```diff
@@ -186,9 +186,12 @@ namespace libtorrent
 		if (m_threads.size() == 0) return;
 		m_idle_timer.expires_from_now(reap_idle_threads_interval);
 		m_idle_timer.async_wait([this](error_code const& e) { reap_idle_threads(e); });
```

Comment:
The lambda happens to be less typing in this case due to the long class name, but mostly I just prefer lambdas. Bind is the quintessential example of a class which produces impenetrable error messages when you make a mistake. I never liked the use of placeholders::_# for unbound arguments either.